src/ui/states/state-needs-payment-info.svelte

```diff
@@ -158,23 +161,41 @@
           brandingInfo?.app_name,
         );
 
+        const linkAuthenticationElement = StripeService.createEmailElement(
+          elements,
+          customerEmail,
+        );
+
+        const triggerError = (error: StripeError) => {
+          isMounted = false;
+          const purchaseError = new PurchaseFlowError(
+            PurchaseFlowErrorCode.ErrorSettingUpPurchase,
+            "Failed to load payment form",
+            error instanceof Error ? error.message : String(error),
+          );
+          handlePaymentError(purchaseError);
+        };
+
         paymentElement.mount("#payment-element");
+        linkAuthenticationElement.mount("#link-authentication-element");
+
+        linkAuthenticationElement.on("change", (event) => {
+          email = event.value.email;
+          isEmailInfoComplete = event.complete;
+        });
+
+        linkAuthenticationElement.on("loaderror", (event) => {
+          triggerError(event.error);
+        });
+
         paymentElement.on(
           "change",
           (event: StripePaymentElementChangeEvent) => {
             isPaymentInfoComplete = event.complete;
           },
         );
         paymentElement.on("loaderror", (event) => {
-          isMounted = false;
-          const purchaseError = new PurchaseFlowError(
-            PurchaseFlowErrorCode.ErrorSettingUpPurchase,
-            "Failed to load payment form",
-            event.error instanceof Error
-              ? event.error.message
-              : String(event.error),
-          );
-          handlePaymentError(purchaseError);
+          triggerError(event.error);
         });
       } catch (error) {
         const purchaseError = new PurchaseFlowError(
```

Comment:
We could also call `triggerError(error)`, as there is no need for a different error message.